### PR TITLE
Api feature

### PR DESCRIPTION
**Pull request by j-tee**
# In This milestone:

-  For each URL create the following:
    - A route.
    - An action in the correct controller.
    - A view file.

-  View files should include only placeholders written in plain HTML, e.g:
    - Here is a list of posts for a given user

-  Make sure that you added code to handle **only** the 4 required URLs.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,7 +1,9 @@
-# frozen_string_literal: true
-
 Rails.application.routes.draw do
-  root 'home#index'
+  root 'users#index'
+  get 'users/:user_id' => 'users#show', :defaults => { format: 'html' }, as: :user
+  get '/users/:user_id/posts/:post_id', to: 'users#user_post', as: 'user_post'
+  get '/users/:user_id/posts', to: 'users#user_posts', as: 'user_posts'
```
Comment (fobadara) [inline config/routes.rb:5]: - You could have done this.
```
resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
```

**Review by fobadara — CHANGES_REQUESTED** [review]
Hi @j-tee, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

## Highlights
✅ Linters are passing

## Required Changes ♻️
_Check the comments under the review._
I had to include some comments here because there are no files to make inline comments in.
[REQUIRED] Kindly include a posts controller with an index and show actions 
[REQUIRED] Kindly include views for posts' actions

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: config/routes.rb (hunk 1 above)
Comment (j-tee) [reply config/routes.rb:5]: Does it mean what I did is wrong

Comment (ShadyShawkat) [inline config/routes.rb:5]: - [ ] [OPTIONAL] Kindly implement the routes like the following as this is a cleaner way of writing it and should be easier to modify in the future.

```ruby 
resources :users, only: [:index, :show] do
   resources :posts, only: [:index, :show]
end

root 'users#index'
```

**Review by ShadyShawkat — APPROVED** [review]
## Hi @j-tee, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight
- All linters are passing ✔️ 
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Lameck1 — COMMENTED** [review]
Hello @j-tee👋,

_This is an additional review as a result of the second opinion request._

Concerning your issue with the first review, you are right. @fobadara could have given the suggestion as OPTIONAL. I am happy that you resubmitted your work and it got approved.

Keep rocking 🚀!

**Pull request merged**